### DISCUSSION TITLE
Python GGUFReader

### DISCUSSION THREAD

**Discussion by Ronsor** [ggml-org/llama.cpp#3718] [Ideas]
The `gguf` Python package has a GGUFWriter class. Why not also a GGUFReader class? It could be useful for converting models back to torch format or inspecting already converted models.

**Comment by monatis** [comment]
Why do you need to convert a GGUF file back to Pytorch --does it have a real-world use case? Obtaining the original Pytorch checkpoints should be much easier than converting them from a GGUF file.

**Reply by Ronsor** [reply]
I mentioned GGUF-to-PyTorch conversion as something that could be done, but my particular use-case is writing tools to dump/inspect GGUF models.

I intend to write the class anyway at some point, but I'd like to get feedback on the idea.

**Comment by KerfuffleV2** [comment]
@ronsor I have a pull open currently to add read/modification support to the Python GGUF package: #3981

If you get a chance, any testing/feedback is very welcome.
